### PR TITLE
User statistics implementation

### PR DESCRIPTION
**Pull request by Ourobore**
Implemented user stats. `User` and `Stats` are different tables that are linked with a bidirectional OneToOne relation: `Stats` store the corresponding `userId`, and `User` should be able to access its stats with the relation.
Statistics are accessed via the `/stats/:id` (or `/stats/:id/played`, `/stats/:id/win`, ...) routes. Added corresponding routes in `/users`: `/users/:id/stats`, `/users/:id/played`, ... that redirect (303: SeeOther) to the `/stats/:id` routes.
Data validation for `:id` not done yet
Added DBeaver, but not using it for now, need further testing for setup automation

### CODE REVIEW

File: backend/src/api/matches/matches.controller.spec.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { MatchesController } from './matches.controller';
+
+describe('MatchesController', () => {
+  let controller: MatchesController;
+
+  beforeEach(async () => {
+    const module: TestingModule = await Test.createTestingModule({
```
Comment (josayko) [inline backend/src/api/matches/matches.controller.spec.ts:8]: Is it for Unit Testing ? Interesting :bulb:

File: backend/src/api/users/entities/user.entity.ts [hunk 2]
```diff
@@ -13,6 +15,9 @@ class User {
 
   @Column()
   public username: string;
+
+  @OneToOne(() => Stats, (stats) => stats.user)
```
Comment (josayko) [inline backend/src/api/users/entities/user.entity.ts:19]: :octopus: ok, this is how each user is linked to his game stats, etc

File: backend/src/auth/auth.service.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import { Injectable } from '@nestjs/common';
+
+@Injectable()
+export class AuthService {
+  getHello(): string {
```
Comment (josayko) [inline backend/src/auth/auth.service.ts:5]: Maybe we should get rid of this now ahah :laughing:

**Review by josayko — APPROVED** [review]
Nothing to report. Lots of files but this is the way :star:

File: backend/src/api/matches/matches.controller.spec.ts (hunk 1 above)
Comment (Ourobore) [reply backend/src/api/matches/matches.controller.spec.ts:8]: Yes I think so. It create the spec file automaticly when you create a controller a service file with the `nest generate` command. Haven't done any research on it yet

File: backend/src/api/users/entities/user.entity.ts (hunk 2 above)
Comment (Ourobore) [reply backend/src/api/users/entities/user.entity.ts:19]: Yes, that is TypeORM way of linking entities in tables (https://typeorm.io/#/one-to-one-relations)

File: backend/src/auth/auth.service.ts (hunk 3 above)
Comment (Ourobore) [reply backend/src/auth/auth.service.ts:5]: One day it will work!

**Comment by Ourobore** [comment]
Yes there is a lot of file for every module, plus some modules that doesn't work / are not fully implemented yet

**Pull request merged**